### ISSUE TITLE
Both current and stable releases of bspump & asab have an incompatible aiokafka requirement

### ISSUE THREAD

**Issue by chinese-soup** [LibertyAces/BitSwanPump#109]
Hello. The stable(?) releases available here (https://github.com/LibertyAces/BitSwanPump/releases) and on PYPI (https://pypi.org/project/bspump/#history) have an offending aiokafka version requirement that gets resolved into a version with a breaking change (0.7.0)

Offending line (bspump-20.7.20.tar.gz/setup.py:71):
```
└──$ nl setup.py  | grep aiokafka -B5 -A5
    66		},
    67		install_requires=[
    68			'asab>=20.6',
    69			'aiohttp>=3.6.2',
    70			'requests>=2.24.0',
    71			'aiokafka>=0.6.0',
    72			'aiozk>=0.25.0',
    73			'aiosmtplib>=1.1.3',
    74			'fastavro>=0.23.5',
    75			'google-api-python-client>=1.7.10',
    76			'numpy>=1.19.0',

```

**Behaviour:**
```
┌──[unko@ryzen]─[~/repos/vevns-test/bitswan-pypi-test/bin]
└──$ . activate
(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test/bitswan-pypi-test/bin]
└──$ cd ../..
(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ pip3 install bspump 
Collecting bspump
  Using cached bspump-20.7.20-py3-none-any.whl (251 kB)
[...]
Collecting aiokafka>=0.6.0
  Using cached aiokafka-0.7.0-cp37-cp37m-manylinux2010_x86_64.whl (1.1 MB)
[...]
Successfully installed aiohttp-3.7.4.post0 aiokafka-0.7.0 aiomysql-0.0.21 aiosmtplib-1.1.4 aiozk-0.28.0 asab-20.7.28 async-timeout-3.0.1 attrs-20.3.0 bspump-20.7.20 cachetools-4.2.1 certifi-2020.12.5 cffi-1.14.5 chardet-4.0.0 cryptography-3.4.6 fastavro-1.3.2 fastjsonschema-2.15.0 google-api-core-1.26.1 google-api-python-client-2.0.2 google-auth-1.27.1 google-auth-httplib2-0.1.0 googleapis-common-protos-1.53.0 httplib2-0.19.0 idna-2.10 kafka-python-2.0.2 mongoquery-1.3.6 motor-2.3.1 multidict-5.1.0 mysql-replication-0.23 netaddr-0.8.0 ntlm-auth-1.5.0 numpy-1.20.1 packaging-20.9 pandas-1.2.3 pika-1.2.0 protobuf-3.15.5 pyarrow-3.0.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.20 pymongo-3.11.3 pymysql-0.9.2 pyparsing-2.4.7 python-dateutil-2.8.1 pytz-2021.1 pywinrm-0.4.1 pyyaml-5.4.1 requests-2.25.1 requests-ntlm-1.1.0 rsa-4.7.2 six-1.15.0 typing-extensions-3.7.4.3 uritemplate-3.0.1 urllib3-1.26.3 xmltodict-0.12.0 yarl-1.6.3
WARNING: You are using pip version 20.1.1; however, version 21.0.1 is available.
You should consider upgrading via the '/home/unko/repos/vevns-test/bitswan-pypi-test/bin/python3.7 -m pip install --upgrade pip' command.

(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ python3 ../../BitSwanPump/examples/bspump-kafka.py 
BitSwan BSPump version 20.07.20
Traceback (most recent call last):
  File "../../BitSwanPump/examples/bspump-kafka.py", line 31, in <module>
    KafkaPipeline(app, "KafkaPipeline")
  File "../../BitSwanPump/examples/bspump-kafka.py", line 12, in __init__
    bspump.kafka.KafkaSource(app, self, "KafkaConnection", config={'topic': 'messages'}),
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 124, in __init__
    self.create_consumer()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 158, in create_consumer
    **self.ConsumerParams
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/connection.py", line 85, in create_consumer
    **kwargs
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/consumer/consumer.py", line 280, in __init__
    sasl_oauth_token_provider=sasl_oauth_token_provider)
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/client.py", line 151, in __init__
    self._md_update_waiter = create_future()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/util.py", line 29, in create_future
    loop = get_running_loop()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/util.py", line 71, in get_running_loop
    "The object should be created within an async function or "
RuntimeError: The object should be created within an async function or provide loop directly.
(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ python3 
Python 3.7.9 (default, Oct  4 2020, 18:12:07) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import asab
>>> import bspump
>>> import aiokafka
>>> asab.__version__
'20.07.28'
>>> bspump.__version__
'20.07.20'
>>> aiokafka.__version__
'0.7.0'
>>> 
>>> 
(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ pip3 install aiokafka==0.6.0
Collecting aiokafka==0.6.0
  Using cached aiokafka-0.6.0.tar.gz (359 kB)
Requirement already satisfied: kafka-python>=2.0.0 in ./bitswan-pypi-test/lib/python3.7/site-packages (from aiokafka==0.6.0) (2.0.2)
Using legacy setup.py install for aiokafka, since package 'wheel' is not installed.
Installing collected packages: aiokafka
  Attempting uninstall: aiokafka
    Found existing installation: aiokafka 0.7.0
    Uninstalling aiokafka-0.7.0:
      Successfully uninstalled aiokafka-0.7.0
    Running setup.py install for aiokafka ... done
Successfully installed aiokafka-0.6.0
WARNING: You are using pip version 20.1.1; however, version 21.0.1 is available.
You should consider upgrading via the '/home/unko/repos/vevns-test/bitswan-pypi-test/bin/python3.7 -m pip install --upgrade pip' command.
(bitswan-pypi-test) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ python3 ../../BitSwanPump/examples/bspump-kafka.py 
BitSwan BSPump version 20.07.20
1 pipeline(s) ready.
08-Mar-2021 15:51:14.115191 ERROR aiokafka Unable connect to "localhost:9092": Multiple exceptions: [Errno 111] Connect call failed ('::1', 9092, 0, 0), [Errno 111] Connect call failed ('127.0.0.1', 9092)
08-Mar-2021 15:51:14.115330 ERROR bspump.pipeline Pipeline 'KafkaPipeline' stopped due to a processing error: KafkaConnectionError: Unable to bootstrap from [('localhost', 9092, <AddressFamily.AF_UNSPEC: 0>)] (<class 'kafka.errors.KafkaConnectionError'>)
Traceback (most recent call last):
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/abc/source.py", line 48, in _main
    await self.main()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 173, in main
    await self.initialize_consumer()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 162, in initialize_consumer
    await self.Consumer.start()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/consumer/consumer.py", line 336, in start
    await self._client.bootstrap()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/client.py", line 233, in bootstrap
    'Unable to bootstrap from {}'.format(self.hosts))
kafka.errors.KafkaConnectionError: KafkaConnectionError: Unable to bootstrap from [('localhost', 9092, <AddressFamily.AF_UNSPEC: 0>)]
08-Mar-2021 15:51:14.115985 ERROR bspump.abc.source Exception in the source 'KafkaSource'
Traceback (most recent call last):
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/abc/source.py", line 48, in _main
    await self.main()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 173, in main
    await self.initialize_consumer()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/source.py", line 162, in initialize_consumer
    await self.Consumer.start()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/consumer/consumer.py", line 336, in start
    await self._client.bootstrap()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/client.py", line 233, in bootstrap
    'Unable to bootstrap from {}'.format(self.hosts))
kafka.errors.KafkaConnectionError: KafkaConnectionError: Unable to bootstrap from [('localhost', 9092, <AddressFamily.AF_UNSPEC: 0>)]
08-Mar-2021 15:51:14.116128 ERROR aiokafka Unable connect to "localhost:9092": Multiple exceptions: [Errno 111] Connect call failed ('::1', 9092, 0, 0), [Errno 111] Connect call failed ('127.0.0.1', 9092)
^CERROR when logging: sys.meta_path is None, Python is likely shutting down
2021-03-08 15:51:15.005951 ERROR root Unclosed AIOKafkaConsumer
{'consumer': <aiokafka.consumer.consumer.AIOKafkaConsumer object at 0x7f2e0f1483d0>}
ERROR when logging: sys.meta_path is None, Python is likely shutting down
2021-03-08 15:51:15.006490 ERROR root Task exception was never retrieved
{'future': <Task finished coro=<KafkaSink._connection() done, defined at /home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/sink.py:155> exception=KafkaConnectionError("Unable to bootstrap from [('localhost', 9092, <AddressFamily.AF_UNSPEC: 0>)]")>}
Traceback (most recent call last):
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/bspump/kafka/sink.py", line 158, in _connection
    await producer.start()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/producer/producer.py", line 287, in start
    await self.client.bootstrap()
  File "/home/unko/repos/vevns-test/bitswan-pypi-test/lib/python3.7/site-packages/aiokafka/client.py", line 233, in bootstrap
    'Unable to bootstrap from {}'.format(self.hosts))
kafka.errors.KafkaConnectionError: KafkaConnectionError: Unable to bootstrap from [('localhost', 9092, <AddressFamily.AF_UNSPEC: 0>)]
```

**Comment by chinese-soup** [comment]
**Correction:** The current tip of the master branch has the same issue. aiokafka>=0.6.0 in setup.py installs 0.7.0 with the breaking change.

```
┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ . test.venvs.git/bin/activate
(test.venvs.git) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ pip3 freeze
(test.venvs.git) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ pip3 install git+https://github.com/LibertyAces/BitSwanPump.git git+https://github.com/TeskaLabs/asab.git
Collecting git+https://github.com/LibertyAces/BitSwanPump.git
  Cloning https://github.com/LibertyAces/BitSwanPump.git to /tmp/pip-req-build-98mt84um
  Running command git clone -q https://github.com/LibertyAces/BitSwanPump.git /tmp/pip-req-build-98mt84um
Collecting git+https://github.com/TeskaLabs/asab.git
  Cloning https://github.com/TeskaLabs/asab.git to /tmp/pip-req-build-j2tr4cba
  Running command git clone -q https://github.com/TeskaLabs/asab.git /tmp/pip-req-build-j2tr4cba
Collecting asab>=20.7.28
  Using cached asab-20.7.28-py3-none-any.whl (74 kB)
[...]
Collecting aiokafka>=0.6.0
  Using cached aiokafka-0.7.0-cp37-cp37m-manylinux2010_x86_64.whl (1.1 MB)
[...]
Using legacy setup.py install for bspump, since package 'wheel' is not installed.
Using legacy setup.py install for mysql-replication, since package 'wheel' is not installed.
Using legacy setup.py install for mongoquery, since package 'wheel' is not installed.
Using legacy setup.py install for pywinrm, since package 'wheel' is not installed.
Installing collected packages: idna, multidict, typing-extensions, yarl, chardet, async-timeout, attrs, aiohttp, fastjsonschema, aiozk, asab, certifi, urllib3, requests, kafka-python, aiokafka, aiosmtplib, fastavro, six, protobuf, googleapis-common-protos, pytz, pyparsing, packaging, cachetools, pyasn1, pyasn1-modules, rsa, google-auth, google-api-core, httplib2, google-auth-httplib2, uritemplate, google-api-python-client, numpy, pika, pycparser, cffi, cryptography, pymysql, aiomysql, mysql-replication, netaddr, pyyaml, pymongo, motor, mongoquery, pybind11, pysimdjson, xmltodict, ntlm-auth, requests-ntlm, pywinrm, python-dateutil, pandas, xxhash, bspump
    Running setup.py install for mysql-replication ... done
    Running setup.py install for mongoquery ... done
    Running setup.py install for pywinrm ... done
    Running setup.py install for bspump ... done
Successfully installed aiohttp-3.7.4.post0 aiokafka-0.7.0 aiomysql-0.0.21 aiosmtplib-1.1.4 aiozk-0.28.0 asab-20.7.28 async-timeout-3.0.1 attrs-20.3.0 bspump-21.01-rc2-161-g0d5dd13 cachetools-4.2.1 certifi-2020.12.5 cffi-1.14.5 chardet-4.0.0 cryptography-3.4.6 fastavro-1.3.2 fastjsonschema-2.15.0 google-api-core-1.26.1 google-api-python-client-2.0.2 google-auth-1.27.1 google-auth-httplib2-0.1.0 googleapis-common-protos-1.53.0 httplib2-0.19.0 idna-2.10 kafka-python-2.0.2 mongoquery-1.3.6 motor-2.3.1 multidict-5.1.0 mysql-replication-0.23 netaddr-0.8.0 ntlm-auth-1.5.0 numpy-1.20.1 packaging-20.9 pandas-1.2.3 pika-1.2.0 protobuf-3.15.5 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybind11-2.6.2 pycparser-2.20 pymongo-3.11.3 pymysql-0.9.2 pyparsing-2.4.7 pysimdjson-3.2.0 python-dateutil-2.8.1 pytz-2021.1 pywinrm-0.4.1 pyyaml-5.4.1 requests-2.25.1 requests-ntlm-1.1.0 rsa-4.7.2 six-1.15.0 typing-extensions-3.7.4.3 uritemplate-3.0.1 urllib3-1.26.3 xmltodict-0.12.0 xxhash-2.0.0 yarl-1.6.3
WARNING: You are using pip version 20.1.1; however, version 21.0.1 is available.
You should consider upgrading via the '/home/unko/repos/vevns-test/test.venvs.git/bin/python3.7 -m pip install --upgrade pip' command.
(test.venvs.git) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ pip3 install --upgrade git+https://github.com/TeskaLabs/asab.git
Collecting git+https://github.com/TeskaLabs/asab.git
  Cloning https://github.com/TeskaLabs/asab.git to /tmp/pip-req-build-wzp_qadt
  Running command git clone -q https://github.com/TeskaLabs/asab.git /tmp/pip-req-build-wzp_qadt
Requirement already satisfied, skipping upgrade: aiohttp>=3.6.2 in ./test.venvs.git/lib/python3.7/site-packages (from asab===21.01-rc2-80-g19a3e0f) (3.7.4.post0)
Requirement already satisfied, skipping upgrade: fastjsonschema>=2.14.4 in ./test.venvs.git/lib/python3.7/site-packages (from asab===21.01-rc2-80-g19a3e0f) (2.15.0)
Requirement already satisfied, skipping upgrade: aiozk>=0.25.0 in ./test.venvs.git/lib/python3.7/site-packages (from asab===21.01-rc2-80-g19a3e0f) (0.28.0)
Requirement already satisfied, skipping upgrade: attrs>=17.3.0 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (20.3.0)
Requirement already satisfied, skipping upgrade: multidict<7.0,>=4.5 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (5.1.0)
Requirement already satisfied, skipping upgrade: async-timeout<4.0,>=3.0 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (3.0.1)
Requirement already satisfied, skipping upgrade: typing-extensions>=3.6.5 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (3.7.4.3)
Requirement already satisfied, skipping upgrade: chardet<5.0,>=2.0 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (4.0.0)
Requirement already satisfied, skipping upgrade: yarl<2.0,>=1.0 in ./test.venvs.git/lib/python3.7/site-packages (from aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (1.6.3)
Requirement already satisfied, skipping upgrade: idna>=2.0 in ./test.venvs.git/lib/python3.7/site-packages (from yarl<2.0,>=1.0->aiohttp>=3.6.2->asab===21.01-rc2-80-g19a3e0f) (2.10)
Using legacy setup.py install for asab, since package 'wheel' is not installed.
ERROR: bspump 21.01-rc2-161-g0d5dd13 has requirement asab>=20.7.28, but you'll have asab 21.01-rc2-80-g19a3e0f which is incompatible.
Installing collected packages: asab
  Attempting uninstall: asab
    Found existing installation: asab 20.7.28
    Uninstalling asab-20.7.28:
      Successfully uninstalled asab-20.7.28
    Running setup.py install for asab ... done
Successfully installed asab-21.01-rc2-80-g19a3e0f
WARNING: You are using pip version 20.1.1; however, version 21.0.1 is available.
You should consider upgrading via the '/home/unko/repos/vevns-test/test.venvs.git/bin/python3.7 -m pip install --upgrade pip' command.
(test.venvs.git) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ python3 ../BitSwanPump/examples/bspump-kafka.py 
BitSwan BSPump version 21.01-rc2-161-g0d5dd13
Traceback (most recent call last):
  File "../BitSwanPump/examples/bspump-kafka.py", line 31, in <module>
    KafkaPipeline(app, "KafkaPipeline")
  File "../BitSwanPump/examples/bspump-kafka.py", line 12, in __init__
    bspump.kafka.KafkaSource(app, self, "KafkaConnection", config={'topic': 'messages'}),
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/bspump/kafka/source.py", line 124, in __init__
    self.create_consumer()
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/bspump/kafka/source.py", line 158, in create_consumer
    **self.ConsumerParams
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/bspump/kafka/connection.py", line 83, in create_consumer
    **kwargs
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/aiokafka/consumer/consumer.py", line 280, in __init__
    sasl_oauth_token_provider=sasl_oauth_token_provider)
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/aiokafka/client.py", line 151, in __init__
    self._md_update_waiter = create_future()
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/aiokafka/util.py", line 29, in create_future
    loop = get_running_loop()
  File "/home/unko/repos/vevns-test/test.venvs.git/lib/python3.7/site-packages/aiokafka/util.py", line 71, in get_running_loop
    "The object should be created within an async function or "
RuntimeError: The object should be created within an async function or provide loop directly.

(test.venvs.git) ┌──[unko@ryzen]─[~/repos/vevns-test]
└──$ python3 
Python 3.7.9 (default, Oct  4 2020, 18:12:07) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import asab, bspump, aiokafka
>>> asab.__version__
'21.01-rc2-80-g19a3e0f'
>>> bspump.__version__
'21.01-rc2-161-g0d5dd13'
>>> aiokafka.__version__
'0.7.0'
>>> 


```

**Comment by ateska** [comment]
Addressed by #110

**Comment by chinese-soup** [comment]
Thanks!
